tiledb/sm/compressors/gzip_compressor.cc

```diff
@@ -143,8 +148,11 @@ void GZip::decompress(
 }
 
 uint64_t GZip::overhead(uint64_t buffer_size) {
-  return 6 + 5 * uint64_t((std::ceil(buffer_size / 16834.0)));
+  // The zlib encoding adds 6 bytes (we don't use compression dictionary)
+  // and the overhead of deflate was taken from
+  // https://stackoverflow.com/a/23578269.
+  return 6 + 5 * uint64_t(std::floor(buffer_size / 16834.0) + 1);
 }
 
-};  // namespace sm
+}  // namespace sm
```

Comment:
Let's take the opportunity to change this to tiledb::sm above instead of having the two seperated.

Reply:
Done.